lib/spack/spack/test/cmd/env.py

```diff
@@ -3122,6 +3124,37 @@ def test_environment_depfile_makefile(depfile_flags, expected_installs, tmpdir,
     assert len(specs_that_make_would_install) == len(set(specs_that_make_would_install))
 
 
+def test_environment_depfile_spec_format_special_chars(mock_packages, monkeypatch):
+    env("create", "test")
+    with ev.read("test"):
+        add("dttop")
+        concretize()
+
+    orig_format = spack.spec.Spec.format
+
+    def _contrived_spec_format(obj, format_str):
+        if obj.name == "dtlink1":
+            return "dtlink1-githash=version-spackhash"
+        else:
+            return orig_format(obj, format_str)
+
+    monkeypatch.setattr(spack.spec.Spec, "format", _contrived_spec_format)
```

Comment:
I'm sorry, but I still don't find this test acceptable. Don't mock very popular internals, they may be relied on elsewhere, and only make refactoring hard. I'll just push an integration test.